Refactoring

mdornaus opened the pull request.

mdornaus committed:
Removed unused Gateway.xml

mdornaus committed:
Fixed decoding of enums in MqttProvider.
Removed Console.Out.WriteLines.
Added some Log Messages.
Extended the Reading of EnumTypes in the Type Dictionary.

mdornaus committed:
Fixed BrowsePathTranslation Request with empty Request Items.
Removed debug function and Replaced it with the appropriate Logger in MqttProvider.
Added xml attribute "includeStructuredComponents" that can be used to include or exclude HasStructuredComponent references in the resulting mqtt format.
Fixed some Linter issues.

mdornaus committed:
Fixed Linter warnings.

Kantiran91 commented on umatiGateway/Configuration/umatiGatewayConfig.xml:
```diff
@@ -1,2 +1,2 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<umatiGatewayConfig version="1.0" autostart="True" file="./Configuration/Files/LocalConfigLocalSampleServer.xml" logLevel="Debug" ReadExtraLibs="False" singleThreadPolling="False" pollTime="1000" />
\ No newline at end of file
+<umatiGatewayConfig version="1.0" autostart="False" file="./Configuration/Files/LocalConfigLocalSampleServer.xml" logLevel="Debug" ReadExtraLibs="False" includeStructuredComponents="True" singleThreadPolling="False" pollTime="1000" />
```
bitte per default sie nicht mitnehmen.

mdornaus replied:
Per default jetzt auf false gesetzt.

Kantiran91 commented on umatiGateway/ConfigurationWriter.cs:
```diff
@@ -56,6 +56,9 @@ public string WriteApplicationConfigToString(Configuration configuration)
             XmlAttribute readExtraLibs = xmlDocument.CreateAttribute("ReadExtraLibs");
             readExtraLibs.Value = configuration.readExtraLibs.ToString();
             umatiGatewayConfigNode.Attributes.Append(readExtraLibs);
+            XmlAttribute includeStructuredComponents = xmlDocument.CreateAttribute("includeSTructuredComponents");
```
includeSTructuredComponents oder includeStructuredComponents

mdornaus replied:
Implemented Changes

Kantiran91 reviewed (changes requested):
I have found to small point. Can you look at them?

mdornaus committed:
Fixed Typo.
Changed config value for include StructuredComponents.
Changed Poll time to 5000.
